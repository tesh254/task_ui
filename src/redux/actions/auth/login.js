import { toast } from "react-toastify";
import axios from "axios";
import { ERROR, LOADING, LOGIN_SUCCESS } from "../../type";

const loginUser = async data => async dispatch => {
  dispatch({
    type: LOADING
  });

  axios
    .post("https://the-task-api.herokuapp.com/auth/login", data)
    .then(res => {
      dispatch({
        type: LOGIN_SUCCESS,
        payload: res.data
      });
      /* istanbul ignore next */
      toast.success("Log in success");
      /* istanbul ignore next */
      setTimeout(() => {
        window.location.href = "/";
      }, 1000);
    })
    .catch(err => {
      dispatch({
        type: ERROR,
        payload: err.response.data
      });
      /* istanbul ignore next */
      toast.error(
        err.response.data.error.message || err.response.data.error.password
      );
    });
};

export default loginUser;
