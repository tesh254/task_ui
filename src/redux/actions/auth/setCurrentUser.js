import setAuthToken from "../../../helpers/setAuthToken";
import { AUTH } from "../../type";

// Set logged in user
export const setCurrentUser = decoded => async dispatch => {
  if (decoded) {
    dispatch({
      type: AUTH,
      payload: {
        isAuthenticated: true
      }
    });
  } else {
    dispatch({
      type: AUTH,
      payload: {
        isAuthenticated: false
      }
    });
  }
};

// Log user out
export const logoutUser = () => dispatch => {
  // Remove token from localStorage
  localStorage.removeItem("jwt_token");
  // Remove auth header for future requests
  setAuthToken(false);
  // Set current user to {} which will set isAuthenticated to false
  dispatch(setCurrentUser(false));
};
