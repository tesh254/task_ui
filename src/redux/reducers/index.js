import { combineReducers } from "redux";
import loginReducer from "./login";
import taskReducer from "./tasks"

const rootReducer = combineReducers({
  login: loginReducer,
  task: taskReducer
});

export default rootReducer;
