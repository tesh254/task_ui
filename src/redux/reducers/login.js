import { LOGIN_SUCCESS, ERROR, LOADING, AUTH } from "../type";

export let initialState = {
  message: "",
  token: "",
  isAuthenticated: false,
  isLoading: false
};

const loginReducer = (state = initialState, action) => {
  switch (action.type) {
    case LOADING:
      return {
        ...state,
        isLoading: true
      };
    case AUTH:
      return {
        isAuthenticated: action.payload.isAuthenticated
      };
    case LOGIN_SUCCESS:
      let token;

      token = action.payload.accessToken;
      localStorage.setItem("jwt_token", token);
      return {
        ...state,
        isAuthenticated: true,
        isLoading: false,
        token
      };
    case ERROR:
      return {
        ...state,
        isLoading: false
      };
    default:
      return state;
  }
};

export default loginReducer;
