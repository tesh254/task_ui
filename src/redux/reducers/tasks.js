import { FETCH_TASKS, LOADING, ERROR } from "../type";

export let initialState = {
  tasks: [],
  isLoading: true,
  page: 1,
  perPage: 5,
  totalTasks: 5
};

const taskReducer = (state = initialState, action) => {
  switch (action.type) {
    case LOADING:
      return {
        ...state,
        isLoading: true
      };
    case FETCH_TASKS:
        console.log(action.payload.totalTasks / action.payload.perPage)
      return {
        ...state,
        tasks: action.payload.tasks,
        isLoading: false,
        page: action.payload.page,
        lastPage: action.payload.totalTasks / action.payload.perPage,
        perPage: action.payload.perPage,
        totalTasks: action.payload.totalTasks
      };
    case ERROR:
      return {
        isLoading: false
      };
    default:
      return state;
  }
};

export default taskReducer;
