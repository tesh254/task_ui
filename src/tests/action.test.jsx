import fetchMock from "fetch-mock";
import { apiMiddleware, ApiError } from "redux-api-middleware";
import configureMockStore from "redux-mock-store";
import loginUser from "../redux/actions/auth/login";
import { setCurrentUser, logoutUser } from "../redux/actions/auth/setCurrentUser"
import { initialState } from "../redux/reducers/login";

const createStore = configureMockStore([apiMiddleware]);
const store = createStore(initialState);

// describe("login actions", () => {
//   afterEach(() => {
//     fetchMock.reset();
//     fetchMock.restore();
//   });

//   it("dispatched user login success", () => {
//     const response = {
//       reset_password: 0,
//       accessToken:
//         "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJleHAiOjM3OTU5OTI2NDcyLCJkYXRhIjoiMDcwNTE4MTcwNyIsImlhdCI6MTU4MTY2MDAwM30.lUNx_3MCdDxzLvXNyPBjZ9nxe09J62F5M8byF7wvPMg",
//       expires_in: "24h"
//     };

//     fetchMock.postOnce("https://the-task-api.herokuapp.com/auth/login", {
//       body: {
//         response
//       }
//     });

//     const expectedActions = [
//       {
//         type: "LOADING"
//       },
//       {
//         type: "LOGIN_SUCESS",
//         payload: response
//       }
//     ];

//     store
//       .dispatch(
//         loginUser({
//           phonenumber: "0705181707",
//           password: "test254"
//         })
//       )
//       .then(() => {
//         expect(store.getActions()).toEqual(expectedActions);
//       });
//   });
// });

describe("setcurrentuser actions", () => {
    it("should test setCurrentUser action function", () => {
        const dispatch = setCurrentUser(true)

        expect(typeof dispatch).toBe("function")
        expect(dispatch).toMatchSnapshot()
    })

    it("should test setCurrentUser action function", () => {
        const dispatch = setCurrentUser(false)

        expect(typeof dispatch).toBe("function")
        expect(dispatch).toMatchSnapshot()
    })

    it("should test setCurrentUser action function", () => {
        const dispatch = logoutUser()

        expect(typeof dispatch).toBe("function")
        expect(dispatch).toMatchSnapshot()
    })
})
