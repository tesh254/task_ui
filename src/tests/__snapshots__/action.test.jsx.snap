// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`setcurrentuser actions should test setCurrentUser action function 1`] = `[Function]`;

exports[`setcurrentuser actions should test setCurrentUser action function 2`] = `[Function]`;

exports[`setcurrentuser actions should test setCurrentUser action function 3`] = `[Function]`;
